src/k8s/pkg/k8sd/setup/kube_apiserver.go

```diff
@@ -52,8 +54,11 @@ func KubeAPIServer(snap snap.Snap, serviceCIDR string, authWebhookURL string, en
 	if err != nil {
 		return fmt.Errorf("failed to open auth-token-webhook.conf: %w", err)
 	}
+
+	microclusterCAPath := filepath.Join(snap.K8sdStateDir(), "cluster.crt")
```

Comment:
let's use `path.Join` like in other places in this file. also, we are not using this anywhere else, so let's inline this below in the struct.